doc/articles/features/open-id-connect.md

```diff
@@ -0,0 +1,206 @@
+{
+	var redirectUri = WebAuthenticationBroker.GetCurrentApplicationCallbackUri().OriginalString;
+
+	// Create options for endpoint discovery
+	var options = new OidcClientOptions
+	{
+		Authority = "https://demo.identityserver.io",
+		ClientId = "interactive.confidential",
+		ClientSecret = "secret",
+		Scope = "openid profile email api offline_access",
+		RedirectUri = redirectUri,
+		PostLogoutRedirectUri = redirectUri,
+		ResponseMode = OidcClientOptions.AuthorizeResponseMode.Redirect,
+		Flow = OidcClientOptions.AuthenticationFlow.AuthorizationCode
+	};
+
+    // Create the client. In production application, this is often created and stored
+    // directly in the Application class.
+	_oidcClient = new OidcClient(options);
+
+    // Invoke Discovery and prepare a request state, containing the nonce.
+    // This is done here to ensure the discovery mecanism is done before
+    // the user clicks on the SignIn button. Since the opening of a web window
+    // should be done during the handling of a user interaction (here it's the button click),
+    // it will be too late to reach the discovery endpoint.
+    // Not doing this could trigger popup blockers mechanisms in browsers.
+	_loginState = await _oidcClient.PrepareLoginAsync();
+    btnSignin.IsEnabled = true;
+
+    // Same for logout url.
+    _logoutUrl = new Uri(await _oidcClient.PrepareLogoutAsync(new LogoutRequest()));
+    btnSignout.IsEnabled = true;
+}
+```
+
+## Step 6 - Proceed to authentication
+
+Add following button handlers:
+
+``` csharp
+private async void SignIn_Clicked(object sender, RoutedEventArgs e)
+{
+    var startUri = new Uri(_loginState.StartUrl);
+
+	// Important: there should be NO await before calling .AuthenticateAsync() - at least
+	// on WebAssembly, in order to prevent trigering the popup blocker mechanisms.
+    var userResult = await WebAuthenticationBroker.AuthenticateAsync(WebAuthenticationOptions.None, startUri);
+ 
+	if(userResult.ResponseStatus != WebAuthenticationStatus.Success)
+    {
+        // Error or user cancellation
+        return;
+    }
+
+    // User authentication process completed successfully.
+    // Now we need to get authorization tokens from the response
+    var authenticationResult = await _oidcClient.ProcessResponseAsync(userResult.ResponseData, _loginState);
+ 	
+ 	if(authenticationResult.IsError)
+    {
+        var errorMessage = authenticationResult.Error;
+        // TODO: do something with error message
+        return;
+    }
+
+ 	// That's completed. Here you have to token, ready to do something
+ 	var token = authenticationResult.AccessToken;
+ 	var refreshToken = authenticationResult.RefreshToken;
+
+	// TODO: make something useful with the tokens
+}
+
+private async void SignOut_Clicked(object sender, RoutedEventArgs e)
+{
+	// Important: there should be NO await before calling .AuthenticateAsync() - at least
+	// on WebAssembly, in order to prevent trigering the popup blocker mechanisms.
```

Comment:
```suggestion
	// Important: there should be no await before calling .AuthenticateAsync() - at least
	// on WebAssembly, in order to prevent triggering the popup blocker mechanisms.
```